industrial_extrinsic_cal/src/check_if_points_in_pic.cpp

```diff
@@ -0,0 +1,349 @@
+﻿//This program checks to see if the camera can see the target at different poses.
+//Assumes a flat rectangular target.
+
+#include <vector>
+#include <industrial_extrinsic_cal/conical_pose_generator.h>
+#include <industrial_extrinsic_cal/targets_yaml_parser.h>
+#include <Eigen/Dense>
+#include <Eigen/Core>
+#include <boost/shared_ptr.hpp>
+#include <industrial_extrinsic_cal/target.h>
+#include <industrial_extrinsic_cal/targets_yaml_parser.h>
+#include <ros/ros.h>
+#include <ros/node_handle.h>
+#include <geometry_msgs/PoseArray.h>
+#include <eigen_conversions/eigen_msg.h>
+#include "applicable_pose_generator/pose_reachability_filter.h"
+
+#include <geometry_msgs/TransformStamped.h>
+#include <geometry_msgs/Twist.h>
+
+#include <industrial_extrinsic_cal/ros_transform_interface.h>
+#include <industrial_extrinsic_cal/check_if_points_in_pic.h>
+
+#include <industrial_extrinsic_cal/camera_yaml_parser.h>
+
+using std::vector;
+using std::string;
+using industrial_extrinsic_cal::Target;
+using YAML::Node;
+
+// local function prototypes
+geometry_msgs::PoseArray pose_filters(geometry_msgs::PoseArray msg2, tf::StampedTransform tf_transform, int image_width, int image_height, EigenSTL::vector_Affine3d AllcameraPoses,Eigen::Vector3d corner_points[4],double fx, double fy, double cx, double cy );
+void imagePoint(Eigen::Vector3d TargetPoint, double fx, double fy, double cx, double cy, double &u, double &v,Eigen::Affine3d cameraPose);
+geometry_msgs::PoseArray create_all_poses(double poseHeight, double spacing_in_z, double angleOfCone, int numberOfStopsForPhotos, Eigen::Vector2d center_point_of_target );
+
+int addingFactorial(int lastAdded);
+Eigen::Vector2d finds_middle_of_target(Eigen::Vector3d corner_points[4], double center_Of_TargetX, double center_Of_TargetY);
+
+int main(int argc, char **argv)
+{
+  ros::init(argc, argv, "node");
+  sleep(2);
+  ros::NodeHandle n("~");
```

Comment:
remove all these hard-coded numbers unless they are just default values that may be overridden by params. Use ros parameters or arguments (argc,argv)